web_api/web_api.py

```diff
@@ -0,0 +1,89 @@
+import os
+import datetime
+import uuid
+import json
+from typing import Dict, Union
+
+from flask import Flask, request, jsonify
+
+app = Flask(__name__)
+UPLOAD_FOLDER = r'C:\Users\mybj2\Desktop\School\Excellenteam\python\final-project-bezJac\uploads'
+OUTPUT_FOLDER = r'C:\Users\mybj2\Desktop\School\Excellenteam\python\final-project-bezJac\outputs'
+app.config['UPLOAD_FOLDER'] = UPLOAD_FOLDER
+app.config['OUTPUT_FOLDER'] = OUTPUT_FOLDER
+
+
+@app.route('/upload', methods=['POST'])
+def upload() -> Dict[str, str]:
+    """
+    Uploads a file to the server.
+    Returns:
+        dict: A JSON response containing the unique identifier (uid) of the uploaded file.
+    """
+    file = request.files['file']
+    if file:
+        uid = str(uuid.uuid4())
+
+        original_filename = file.filename
+        file_name, file_ext = os.path.splitext(original_filename)
+
+        timestamp = datetime.datetime.now().strftime("%Y%m%d%H%M%S")
+        new_filename = f"{uid}_{timestamp}_{file_name}_{file_ext}"
+
+        file.save(os.path.join(app.config['UPLOAD_FOLDER'], new_filename))
+
+        return jsonify({'uid': uid})
+    else:
+        return jsonify({'error': 'No file provided'}), 400
+
+
+@app.route('/status/<string:uid>', methods=['GET'])
+def status(uid: str) -> Union[Dict[str, Union[str, int, Dict[str, Union[str, int, None]]]], Dict[str, str]]:
```

Comment:
Separate code to small functions.  Read more: https://softwareengineering.stackexchange.com/questions/308108/when-is-it-appropriate-to-make-a-separate-function-when-there-will-only-ever-be